### PR TITLE
I1700 - Kodiak install script

### PR DESCRIPTION
**Pull request by littledice**
Kodiak installation script and Kotlin config parsing

`./build-app.sh`  tags and pushes the Kodiak image (and is running on TC http://teamcity.montagu.dide.ic.ac.uk:8111/viewType.html?buildTypeId=montagu_Backup_Kodiak&branch_montagu_Backup=i1700&tab=buildTypeStatusDiv)

### CODE REVIEW

File: scripts/create-test-config.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+#Copies test config to /etc/kodiak and must be run as sudo
+set -e
+
+here=$(dirname $0)
+
+mkdir -p /etc/kodiak
+cp $here/../config/testconfig.json /etc/kodiak/
```
Comment (MartinEden) [inline scripts/create-test-config.sh]: Why does this have to live at any particular path? Couldn't it just be a local folder and no sudo required?

File: src/main/kotlin/org/vaccineimpact/kodiak/Config.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    val workingPath: String;
+    val targets: List<Target>
+}
+
+data class Target(val id: String,
+                  val encrypted: Boolean,
+                  val remoteBucket: String,
+                  val localPath: String)
+
+class Config(configPath: String = "/etc/kodiak/config.json") : ConfigWrapper {
```
Comment (MartinEden) [inline src/main/kotlin/org/vaccineimpact/kodiak/Config.kt]: Normally I'd expect a concrete class to have a more specific name than the interface. e.g. interface could be `Config` and class could be `JsonConfig`.

File: src/main/kotlin/org/vaccineimpact/kodiak/Config.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        val file = File(configPath)
+        if (!file.exists()) {
+            throw FileNotFoundException("No config found at $configPath")
+        }
+        properties = JsonParser()
+                .parse(InputStreamReader(file.inputStream()))
+                .asJsonObject
+    }
+
+    private fun get(key: String): JsonElement {
```
Comment (MartinEden) [inline src/main/kotlin/org/vaccineimpact/kodiak/Config.kt]: Consider adding the operator keyword here, and then you can use `this["starport_path"]` etc. below.

File: scripts/create-test-config.sh (hunk 1 above)
Comment (littledice) [reply scripts/create-test-config.sh]: you're right! oversight

Comment (littledice) [reply scripts/create-test-config.sh]: Actually we don't need this at all; why not just have a test config file in the test resources folder. Have removed/simplified accordingly.

File: scripts/build-app.sh [hunk 4]
```diff
@@ -1 +1,22 @@
...
+docker build --tag kodiak-build \
+	--build-arg git_id=$git_id \
+	--build-arg git_branch=$git_branch \
+	.
+
+# Run the created image
+docker run \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v $docker_auth_path:/root/.docker/config.json \
+    --network=host \
```
Comment (richfitz) [inline scripts/build-app.sh]: no need for host networking?

File: src/main/kotlin/org/vaccineimpact/kodiak/Config.kt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            return x
+        } else {
+            throw MissingConfigurationKey(key)
+        }
+    }
+
+    override val starportPath: String = this["starport_path"].asString
+    override val workingPath: String = this["working_path"].asString
+
+    override val targets: List<Target> = gson
```
Comment (richfitz) [inline src/main/kotlin/org/vaccineimpact/kodiak/Config.kt]: can this be entirely deserialised using gson

Comment (littledice) [reply src/main/kotlin/org/vaccineimpact/kodiak/Config.kt]: Gonna leave as is for now, but maybe

**Pull request merged**